experiments/AtmosGCM/GCMDriver/GCMDriver.jl

```diff
@@ -0,0 +1,385 @@
+        return nothing
+    else
+        return lowercase(arg)
+    end
+end
+
+# parse `--experiment` command line argument
+function parse_experiment_arg(arg)
+    if arg == "baroclinic_wave"
+        problem_type = BaroclinicWaveProblem
+    elseif arg == "heldsuarez"
+        problem_type = HeldSuarezProblem
+    else
+        error("unknown experiment: " * arg)
+    end
+
+    return problem_type
+end
+
+# experiment configuration setup
+function config_gcm_experiment(
+    FT,
+    poly_order,
+    resolution,
+    experiment_arg,
+    wind_perturbation_arg,
+    base_state_arg,
+    moisture_profile_arg,
+    surface_flux_arg,
+)
+    # Set up a reference state for linearization of equations
+    temp_profile_ref =
+        DecayingTemperatureProfile{FT}(param_set, FT(275), FT(75), FT(45e3))
+    ref_state = HydrostaticState(temp_profile_ref)
+
+    # Distance between surface and top of atmosphere (m)
+    domain_height::FT = 30e3
+
+    # make the orientation explicit (for surface fluxes)
+    orientation = SphericalOrientation()
+
+    # Determine the problem type
+    problem_type = parse_experiment_arg(experiment_arg)
+
+    # Set up problem components
+    wind_perturbation = parse_wind_perturbation_arg(wind_perturbation_arg)
+    base_state = parse_base_state_arg(base_state_arg)
+    moisture_profile = parse_moisture_profile_arg(moisture_profile_arg)
+
+    # Choose the moisture model
+    if moisture_profile isa NoMoistureProfile
+        hyperdiffusion = DryBiharmonic(FT(8 * 3600))
+        moisture = DryModel()
+    else
+        hyperdiffusion = EquilMoistBiharmonic(FT(8 * 3600))
+        moisture = EquilMoist{FT}()
+    end
+
+    # Set up the boundary conditions
+    boundarycondition = parse_surface_flux_arg(
+        surface_flux_arg,
+        FT,
+        param_set,
+        orientation,
+        moisture,
+    )
+
+    # Set up the problem
+    problem = problem_type(
+        boundarycondition = boundarycondition,
+        wind_perturbation = wind_perturbation,
+        base_state = base_state,
+        moisture_profile = moisture_profile,
+    )
+
+    # Create the Atmosphere model
+    model = AtmosModel{FT}(
+        AtmosGCMConfigType,
+        param_set;
+        problem = problem,
+        orientation = orientation,
+        ref_state = ref_state,
+        turbulence = ConstantViscosityWithDivergence(FT(0)),
+        hyperdiffusion = hyperdiffusion,
+        moisture = moisture,
+        source = setup_source(problem),
+    )
+
+    # Create the GCM driver configuration
+    config = ClimateMachine.AtmosGCMConfiguration(
+        problem_name(problem),
+        poly_order,
+        resolution,
+        domain_height,
+        param_set,
+        init_gcm_experiment!;
+        model = model,
+    )
+
+    return config
+end
+
+# default diagnostics configuration setup
+function config_diagnostics(FT, driver_config)
+    interval = "40000steps"
+
+    _planet_radius = planet_radius(param_set)::FT
```

Comment:
What's the difference between this and
```julia
    _planet_radius::FT = planet_radius(param_set)
```
?

Reply:
See [here](https://github.com/CliMA/ClimateMachine.jl/pull/1447#discussion_r474313647); I haven't gotten to the bottom of it yet. Will be exploring that soon.